flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.concurrent;
+
+import org.apache.flink.runtime.concurrent.impl.FlinkCompletableFuture;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+
+public class FutureUtils {
```

Comment:
Yes I've also seen this class. Unfortunately, this class is in flink-core and the Flink futures are in flink-runtime. Unless we move the Flink futures into flink-core or the `FutureUtil` to flink-runtime, we won't be able to merge it. The `FutureUtil` is used in flink-streaming-java. At the moment this module has a dependency on flink-runtime. Thus, the latter approach would work. However, I think that the dependency of flink-streaming-java on flink-runtime should actually not exist (even though this might only exist in an utopian world ;-)

We cannot move the future interfaces alone to flink-core, because `FutureUtils` needs an `CompletableFuture` implementation. Thus, moving the flink futures to flink-core would entail that flink-core gets a dependency on Scala. Also not ideal, imo.